fvm/src/call_manager/default.rs

```diff
@@ -245,13 +245,14 @@ where
     fn with_transaction(
         &mut self,
         f: impl FnOnce(&mut Self) -> Result<InvocationResult>,
+        always_revert: bool,
     ) -> Result<InvocationResult> {
         self.state_tree_mut().begin_transaction();
         self.events.begin_transaction();
         self.state_access_tracker.begin_transaction();
 
         let (revert, res) = match f(self) {
-            Ok(v) => (!v.exit_code.is_success(), Ok(v)),
+            Ok(v) => (always_revert || !v.exit_code.is_success(), Ok(v)),
```

Comment:
I would move this down to a separate line `let revert = always_revert || revert;`

Reply:
I mean technically I feel the current implementation is the right place for the condition. But I think where you are coming from is for better readability? I will update accordingly.

Reply:
Code is written once, and read 100 times. The above forces me to evaluate each branch condition before I can be  certain that `always_revert` does what's on the tin.

Reply:
updated and resolve.